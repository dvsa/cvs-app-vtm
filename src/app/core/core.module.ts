import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
