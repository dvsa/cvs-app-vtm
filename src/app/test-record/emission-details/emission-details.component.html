<dl *ngIf="!editState" class="govuk-summary-list">
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Emission standard
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.emissionStandard | DefaultNullOrEmpty}}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Smoke test "K" limit applied (m <sup>1</sup>)
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.smokeTestKLimitApplied | DefaultNullOrEmpty}}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Fuel type
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.fuelType | DefaultNullOrEmpty}}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Modification type
    </dt>
    <dd class="govuk-summary-list__value">
      <ng-container *ngIf="testType.modType else hyphen">
        {{(testType.modType.code ? testType.modType.code : '') + ' ' + (testType.modType.description ? testType.modType.description : '') | DefaultNullOrEmpty}}
      </ng-container>
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Modification type used
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.modificationTypeUsed | DefaultNullOrEmpty}}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Particulate trap fitted
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.particulateTrapFitted | DefaultNullOrEmpty}}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Particulate trap serial number
    </dt>
    <dd class="govuk-summary-list__value">
      {{testType.particulateTrapSerialNumber | DefaultNullOrEmpty}}
    </dd>
  </div>
</dl>
<ng-template #hyphen>
  <span>-</span>
</ng-template>

<ng-container *ngIf="editState && testRecord.testResult=='pass'">
  <vtm-emission-details-edit [testType]="testType"></vtm-emission-details-edit>
</ng-container>
