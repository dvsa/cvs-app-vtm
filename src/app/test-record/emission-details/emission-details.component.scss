.govuk-summary-list {
  & .govuk-summary-list__row:last-child {
    & .govuk-summary-list__value, .govuk-summary-list__key {
      border-bottom: none !important;
    }
  }
}
