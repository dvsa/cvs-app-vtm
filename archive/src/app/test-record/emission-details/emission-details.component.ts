import { Component, ChangeDetectionStrategy, Input, OnChanges } from '@angular/core';
import { TestType } from '@app/models/test.type';
import { VIEW_STATE } from '@app/app.enums';
import { TestResultModel } from '@app/models/test-result.model';

@Component({
  selector: 'vtm-emission-details',
  templateUrl: './emission-details.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class EmissionDetailsComponent implements OnChanges {
  @Input() testType: TestType;
  @Input() testRecord: TestResultModel;
  @Input() editState: VIEW_STATE;
  @Input() hasEmissionApplicable: boolean;
  editModTypeUsed: boolean;
  editParticulate: boolean;
  modTypeValue: string;

  constructor() {}

  ngOnChanges() {
    this.modTypeValue = !!this.testType.modType
      ? this.testType.modType.code + ' - ' + this.testType.modType.description
      : '';

    this.editParticulate = this.modTypeValue === 'p - particulate trap';
    this.editModTypeUsed =
      this.modTypeValue === 'g - gas engine' ||
      this.modTypeValue === 'm - modification or change of engine';
  }
}
