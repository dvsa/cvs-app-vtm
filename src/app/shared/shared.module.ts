import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { DocumentRetrievalService } from '@api/document-retrieval';
import { RoleRequiredDirective } from '@directives/app-role-required.directive';
import { AccordionControlComponent } from './components/accordion-control/accordion-control.component';
import { AccordionComponent } from './components/accordion/accordion.component';
import { BannerComponent } from './components/banner/banner.component';
import { BaseDialogComponent } from './components/base-dialog/base-dialog.component';
import { ButtonGroupComponent } from './components/button-group/button-group.component';
import { ButtonComponent } from './components/button/button.component';
import { IconComponent } from './components/icon/icon.component';
import { NumberPlateComponent } from './components/number-plate/number-plate.component';
import { PaginationComponent } from './components/pagination/pagination.component';
import { TagComponent } from './components/tag/tag.component';
import { TestCertificateComponent } from './components/test-certificate/test-certificate.component';
import { RetrieveDocumentDirective } from './directives/retrieve-document/retrieve-document.directive';
import { PreventDoubleClickDirective } from './directives/prevent-double-click/prevent-double-click.directive';
import { DefaultNullOrEmpty } from './pipes/default-null-or-empty/default-null-or-empty.pipe';
import { DigitGroupSeparatorPipe } from './pipes/digit-group-separator/digit-group-separator.pipe';
import { RefDataDecodePipe } from './pipes/ref-data-decode/ref-data-decode.pipe';
import { TestTypeNamePipe } from './pipes/test-type-name/test-type-name.pipe';

@NgModule({
  declarations: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    PreventDoubleClickDirective,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  imports: [CommonModule, RouterModule],
  exports: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  providers: [DocumentRetrievalService]
})
export class SharedModule {}
