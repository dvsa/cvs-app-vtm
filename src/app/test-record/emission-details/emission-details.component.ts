import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TestType } from '@app/models/test.type';
import { VIEW_STATE } from '@app/app.enums';
import { TestResultModel } from '@app/models/test-result.model';

@Component({
  selector: 'vtm-emission-details',
  templateUrl: './emission-details.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class EmissionDetailsComponent implements OnInit {
  @Input() testType: TestType;
  @Input() testRecord: TestResultModel;
  @Input() editState: VIEW_STATE;
  @Input() hasEmissionApplicable: boolean;

  constructor() {}

  ngOnInit() {}
}
