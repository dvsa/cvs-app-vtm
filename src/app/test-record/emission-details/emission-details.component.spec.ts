import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { EmissionDetailsComponent } from './emission-details.component';
import { SharedModule } from '@app/shared/shared.module';
import { TestType } from '@app/models/test.type';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

describe('EmissionDetailsComponent', () => {
  let component: EmissionDetailsComponent;
  let fixture: ComponentFixture<EmissionDetailsComponent>;
  const testType = {} as TestType;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [EmissionDetailsComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(EmissionDetailsComponent);
    component = fixture.componentInstance;
    component.testType = testType;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
