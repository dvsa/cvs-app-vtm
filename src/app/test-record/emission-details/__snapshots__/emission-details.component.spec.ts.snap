// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmissionDetailsComponent should create 1`] = `
<vtm-emission-details
  testType={[Function Object]}
>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Emission standard 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Smoke test "K" limit applied (m 
        <sup>
          1
        </sup>
        )
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Fuel type 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Modification type 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
        
        <span>
          -
        </span>
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Modification type used 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Particulate trap fitted 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
         Particulate trap serial number 
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
         - 
      </dd>
    </div>
  </dl>
</vtm-emission-details>
`;
